src/test/spec/crud_v1/aggregate.rs

```diff
@@ -19,6 +19,7 @@ struct Arguments {
 
 #[function_name::named]
 async fn run_aggregate_test(test_file: TestFile) {
+    let _guard: RwLockReadGuard<()> = LOCK.run_concurrently().await;
```

Comment:
Because we're setting failpoints in tests, acquiring the lock needs to be done before any i/o is performed whatsoever (e.g. in `TestClient::new` or in the background threads of a client). To that end, I updated many tests who were in violation of this, since they were interfering with the failpoints set in the tests required for this work.